doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java

```diff
@@ -54,59 +51,67 @@
      */
-    public DynamoDBDoiRequestsService(Table table, Index index) {
+    protected DynamoDBDoiRequestsService(Table table, Index index) {
         this.objectMapper = JsonUtils.objectMapper;
         this.publicationsTable = table;
         this.doiRequestsIndex = index;
         this.clockForTimestamps = Clock.systemDefaultZone();
     }
 
-    public DynamoDBDoiRequestsService(AmazonDynamoDB client, Environment environment) {
-        this(client, environment, Clock.systemDefaultZone());
-    }
-
-    public DynamoDBDoiRequestsService(AmazonDynamoDB client, Environment environment,
-                                      Clock clockForTimestamps) {
+    protected DynamoDBDoiRequestsService(AmazonDynamoDB client, Environment environment, Clock clockForTimestamps) {
 
-        DynamoDB dynamoDB = new DynamoDB(client);
-        this.publicationsTable = dynamoDB
-            .getTable(environment.readEnv(ServiceConstants.PUBLICATIONS_TABLE_NAME_ENV_VARIABLE));
-        this.doiRequestsIndex = publicationsTable
-            .getIndex(environment.readEnv(ServiceConstants.DOI_REQUESTS_INDEX_ENV_VARIABLE));
         this.clockForTimestamps = clockForTimestamps;
         this.objectMapper = JsonUtils.objectMapper;
-    }
 
-    @JacocoGenerated
-    public static DynamoDBDoiRequestsService defaultDoiRequestService(Environment environment) {
-        return new DynamoDBDoiRequestsService(AmazonDynamoDBClientBuilder.defaultClient(), environment);
+        DynamoDB dynamoDB = new DynamoDB(client);
+        this.publicationsTable = dynamoDB.getTable(
+                environment.readEnv(ServiceConstants.PUBLICATIONS_TABLE_NAME_ENV_VARIABLE));
+        this.doiRequestsIndex = publicationsTable.getIndex(
+                environment.readEnv(ServiceConstants.DOI_REQUESTS_INDEX_ENV_VARIABLE));
     }
 
+
     @Override
     public List<Publication> findDoiRequestsByStatus(URI publisher, DoiRequestStatus status)
         throws ApiGatewayException {
-        return attempt(() -> limitKeyRangeToStatus(status))
-            .map(statusLimitedRange -> queryByPublisherAndStatus(publisher, statusLimitedRange))
+        return
+            extractMostRecentVersionOfEachPublication(publisher)
+                .stream()
+                .filter(publication -> hasDoiRequestStatus(publication, status))
+                .collect(Collectors.toList());
+    }
+
+    private List<Publication> extractMostRecentVersionOfEachPublication(URI publisher) throws DynamoDBException {
+        return attempt(() -> queryByPublisher(publisher))
             .map(this::extractPublications)
             .map(this::keepMostRecentPublications)
             .orElseThrow(this::handleErrorFetchingPublications);
     }
 
+    private boolean hasDoiRequestStatus(Publication publ, DoiRequestStatus status) {
+        return Optional.of(publ)
+            .map(Publication::getDoiRequest)
+            .map(DoiRequest::getStatus)
+            .filter(st -> st.equals(status))
```

Comment:
DoiRequestStatus::equals

Reply:
DoiRequestStatus::equals is equivalent to st->st.equals(st) which is comparing to itself. I am comparing st to the method parameter.

Reply:
Actually it could be fixed if i flipped the arguments